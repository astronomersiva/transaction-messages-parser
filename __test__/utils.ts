import path from 'path';
import { fileURLToPath } from 'url';

export const getFileName = (importMetaUrl: string) => {
  const __filename = fileURLToPath(importMetaUrl);
  return path.basename(__filename);
}
