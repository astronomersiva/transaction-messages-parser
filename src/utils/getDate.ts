export default function getDate(dateString: string | undefined): string | null {
  if (!dateString) {
    return null;
  }

  const parts = dateString.split('-').map(part => part.split(':')).flat();

  if (parts.length === 2) {
    parts.push(new Date().getFullYear().toString().slice(2));
  }

  const date = new Date();
  date.setMonth(parseInt(parts[1]) - 1, parseInt(parts[0]));
  date.setFullYear(parts[2].length === 2 ? 2000 + parseInt(parts[2]) : parseInt(parts[2]));

  const hours = parseInt(parts[3]) || 0;
  const minutes = parseInt(parts[4]) || 0;
  const seconds = parseInt(parts[5]) || 0;

  date.setHours(hours);
  date.setMinutes(minutes);
  date.setSeconds(seconds);

  return date.toString();
}
