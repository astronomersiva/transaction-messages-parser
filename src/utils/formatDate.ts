import getDate from './getDate.js';

const months = {
  'jan': '01',
  'feb': '02',
  'mar': '03',
  'apr': '04',
  'may': '05',
  'jun': '06',
  'jul': '07',
  'aug': '08',
  'sep': '09',
  'oct': '10',
  'nov': '11',
  'dec': '12',
  'january': '01',
  'february': '02',
  'march': '03',
  'april': '04',
  // 'may': '05', already exists
  'june': '06',
  'july': '07',
  'august': '08',
  'september': '09',
  'october': '10',
  'november': '11',
  'december': '12',
};

function getMonthNumber(month: keyof typeof months): typeof months[keyof typeof months]{
  return months[month];
}

export function formatDateWithMonth(date: string, month: string, year: string, hours?: string, mins?: string): string | null {
  const monthNumber = getMonthNumber(month.toLowerCase() as keyof typeof months);
  return getDate(`${date}-${monthNumber}-${year}${hours ? `:${hours}:${mins}` : ''}`);
}

export function formatDateStringWithMonth(dateString: string): string | null {
  const parts = dateString.split('-');
  return formatDateWithMonth(parts[0], parts[1], parts[2]);
}
