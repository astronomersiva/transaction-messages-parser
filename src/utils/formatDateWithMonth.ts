import getDate from './getDate.js';

const months = {
  'jan': '01',
  'feb': '02',
  'mar': '03',
  'apr': '04',
  'may': '05',
  'jun': '06',
  'jul': '07',
  'aug': '08',
  'sep': '09',
  'oct': '10',
  'nov': '11',
  'dec': '12',
};

function getMonthNumber(month: keyof typeof months): typeof months[keyof typeof months]{
  return months[month];
}

export default function formatDateWithMonth(dateString: string): string | null {
  const parts = dateString.split('-');
  const date = parts[0];
  const month = getMonthNumber(parts[1].toLowerCase() as keyof typeof months);
  const year = parts[2];

  return getDate(`${date}-${month}-${year}`);
}
