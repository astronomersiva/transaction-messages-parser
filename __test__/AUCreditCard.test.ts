const { getTransactionData } = await import('../src/index.js');
import fs from 'fs';

import { getFileName } from './utils.js';
const testCases = fs.readFileSync(`__test__/fixtures/${getFileName(import.meta.url).replace('.test.ts', '')}`).toString().split('-----\n');

test('main', async() => {
  for (const testCase of testCases) {
    expect(await getTransactionData(testCase)).toMatchSnapshot();
  }
});
