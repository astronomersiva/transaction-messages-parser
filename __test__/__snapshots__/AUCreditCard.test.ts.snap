// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main 1`] = `
{
  "account": {
    "number": "xx1857",
    "number_formatted": "x1857",
    "type": "card",
  },
  "transaction": {
    "amount": 417.38,
    "currency": "INR",
    "date": "Tue Jan 23 2024 18:38:00 GMT+0530 (India Standard Time)",
    "merchant": "BOOKMYSHOW",
    "type": "debit",
  },
}
`;

exports[`main 2`] = `
{
  "account": {
    "number": "xx1857",
    "number_formatted": "x1857",
    "type": "card",
  },
  "transaction": {
    "amount": 417.38,
    "currency": "INR",
    "date": "Tue Jan 23 2024 00:38:00 GMT+0530 (India Standard Time)",
    "merchant": "BOOKMYSHOW",
    "type": "debit",
  },
}
`;

exports[`main 3`] = `
{
  "account": {
    "number": "xx1857",
    "number_formatted": "x1857",
    "type": "card",
  },
  "transaction": {
    "amount": 417.38,
    "currency": "INR",
    "date": "Tue Jan 23 2024 12:38:00 GMT+0530 (India Standard Time)",
    "merchant": "BOOKMYSHOW",
    "type": "debit",
  },
}
`;
